lib/call.ts

```diff
@@ -2,6 +2,25 @@ import type { Operation } from "./types.ts";
 import { action } from "./instructions.ts";
 import { pause } from "./pause.ts";
 
+function isFunc(f: unknown): boolean {
+  return typeof f === "function";
+}
+function isPromise(p: unknown): boolean {
+  if (!p) return false;
+  return isFunc((p as PromiseLike<unknown>).then);
+}
```

Comment:
Can we use guards where possible? It saves a little casting when you use them.

```suggestion
function isPromise<T>(p: unknown): p is Promise<T> {
  if (!p) return false;
  return isFunc((p as Promise<T>).then);
}
```